### PR TITLE
APIGOV-26573 - create consumer for managed applications

### PR DESCRIPTION
**Pull request by jcollins-axway**

### CODE REVIEW

**Review by vivekschauhan — APPROVED** [review]

**Review by dgghinea — APPROVED** [review]
I think the target branch can be changed to 26463, because I merged the changes from 26570 into that one

File: pkg/subscription/application/application_test.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+func (m mockApplicationRequest) GetManagedApplicationName() string {
+	return m.name
+}
+
+func (m mockApplicationRequest) GetTeamName() string {
+	return m.team
+}
+
+func (m mockApplicationRequest) GetID() string {
```
Comment (dgghinea) [inline pkg/subscription/application/application_test.go:60]: I believe mockApplicationRequest needs also to implement GetConsumerOrgID

Comment (jcollins-axway) [reply pkg/subscription/application/application_test.go:60]: thanks, was using an old sdk before that was added

**Review by alrosca — APPROVED** [review]

File: pkg/subscription/application/application.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	}
+	return a
+}
+
+func (a AppProvisioner) Provision() provisioning.RequestStatus {
+	a.logger.Info("provisioning application")
+
+	rs := provisioning.NewRequestStatusBuilder()
+	if a.appName == "" {
+		log.Error("could not find the managed application name on the resource")
```
Comment (sbolosan) [inline pkg/subscription/application/application.go:70]: a.logger?

File: pkg/subscription/application/application.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	return rs.Success()
+}
+
+func (a AppProvisioner) Deprovision() provisioning.RequestStatus {
+	a.logger.Info("deprovisioning application")
+
+	rs := provisioning.NewRequestStatusBuilder()
+
+	if a.consumerID == "" {
+		log.Error("could not find the consumer id on the managed application resource")
```
Comment (sbolosan) [inline pkg/subscription/application/application.go:92]: a.logger here too?

File: pkg/beater/custom_beater.go [hunk 4]
```diff
@@ -62,7 +62,7 @@ func (bt *customLogBeater) Run(b *beat.Beat) error {
 			defer r.Body.Close()
 
 			if err != nil {
-				fmt.Errorf("Error while reading request body: %s", err)
+				logp.Error(fmt.Errorf("error while reading request body: %s", err))
```
Comment (sbolosan) [inline pkg/beater/custom_beater.go:65]: are you guys meaning to use logp?

Comment (jcollins-axway) [reply pkg/beater/custom_beater.go:65]: not worried about the ta just yet, will address that once we get there

**Review by sbolosan — APPROVED** [review]
TA aside, looks good

**Pull request merged**